sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/ViewFactory.java

```diff
@@ -32,13 +35,35 @@ public View create(StreamModel model, SpiHelper spiHelper, List<Closeable> close
     if (model.getDescription() != null) {
       builder.setDescription(model.getDescription());
     }
-    if (model.getAttributeKeys() != null) {
-      builder.setAttributeFilter(new HashSet<>(model.getAttributeKeys()));
+    IncludeExcludeModel attributeKeys = model.getAttributeKeys();
+    if (attributeKeys != null) {
+      addAttributeKeyFilter(builder, attributeKeys.getIncluded(), attributeKeys.getExcluded());
     }
     if (model.getAggregation() != null) {
       builder.setAggregation(
           AggregationFactory.getInstance().create(model.getAggregation(), spiHelper, closeables));
     }
     return builder.build();
   }
+
+  private static void addAttributeKeyFilter(
+      ViewBuilder builder, @Nullable List<String> included, @Nullable List<String> excluded) {
+    if (included == null && excluded == null) {
+      return;
+    }
+    if (included == null) {
+      Set<String> excludedKeys = new HashSet<>(excluded);
+      // TODO: set predicate with useful toString implementation
+      builder.setAttributeFilter(attributeKey -> !excludedKeys.contains(attributeKey));
+      return;
+    }
+    if (excluded == null) {
+      Set<String> includedKeys = new HashSet<>(included);
```

Comment:
Some question as above about set.

Reply:
In this case the `ViewBuilder` API has an overload for setting an allowlist of attributes to retain, and the API requires `Set<String>`.